### ISSUE TITLE
JavaScript Support

### ISSUE THREAD

**Issue by ideadapt** [fabsx00/joern#53]
In my fork I started to teach joern ECMAScript5 (JavaScript).

If ever, I want to create a PR, which branch should I branch off now, dev or master?

**Comment by fabsx00** [comment]
oh, that's nice to hear. Use the master branch please. I am going to remove the dev branch soon.

**Comment by ideadapt** [comment]
ok great. 
Another thing, are there any plans to introduce a dependency management such as maven? I recognized that some of your references in .classpath are out of date. Is this on purpose, or is it ok to upgrade them?

**Comment by fabsx00** [comment]
No, there are currently no plans to introduce something like maven. If by "out of date" you mean not equal to the newest version of neo2j, that's correct. If the .classpath references and the references in build.xml or the supplied dependency tarball are out of sync, that's a bug. Is that the case? Seems alright at least in the master branch...

**Comment by ideadapt** [comment]
The dependencies were consistent, just behind latest releases. Thanks for
updating neo4j. I'm working on a PR to migrate to antlr  4.5.

2015-04-18 21:00 GMT+02:00 Fabian Yamaguchi <notifications@github.com>:

> No, there are currently no plans to introduce something like maven. If by
> "out of date" you mean not equal to the newest version of neo2j, that's
> correct. If the .classpath references and the references in build.xml or
> the supplied dependency tarball are out of sync, that's a bug. Is that the
> case? Seems alright at least in the master branch...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fabsx00/joern/issues/53#issuecomment-94191757>.
>



-- 
http://www.ideadapt.net - where ideas become code

**Comment by AntoineHus** [comment]
Hello Gentlemen, 

I updated the build.xml and I added a .travis.yml for testing the building process. Now, I'm on gradle and docker for being able to provider a docker for joern directly. 
Are you still working on Joern ?

**Comment by fabsx00** [comment]
Hi!
Yes, we are still working on joern. Most of the recent commits are in 'dev' and 'phpsupport'. Personally, I am currently working on merging joern and bjoern, but that will still take a bit.
Fabian

**Comment by ideadapt** [comment]
Hi Antoine

I am currently not working on the JavaScript support thing. Though it would still be a very good thing to have.
It felt like to much work for me as a hobby project. Another thing was, that ES6 is rising, but I did not find any antlr grammar for ES6.
